File: DesignPatternsExercises/java/FlyweightDesignPattern/Exercise2/Flight.java
public class Flight {
    private String company;

    public Flight(String company) {
        this.company = company;
    }

    public void __________(1)__________(int flightNumber, double latitude, double longitude) {
        System.out.println(company + " flight " + flightNumber + " is currently at latitude " + latitude +
                " and longitude " + longitude);
    }
}

File: DesignPatternsExercises/ruby/flyweight_design_pattern/flyweight_design_pattern2/solution/flight.rb
class Flight
  # def __________(1)__________(company)
  def initialize(company)
    @company = company
  end

  # def __________(2)__________(flight_number, latitude, longitude)
  def draw_on_map(flight_number, latitude, longitude)
    puts "#{@company} flight #{flight_number} is currently at latitude #{latitude} and #{longitude} longitude"
  end
end
